spec/models/preview_card_spec.rb

```diff
@@ -3,6 +3,12 @@
 require 'rails_helper'
 
 RSpec.describe PreviewCard do
+  describe 'file size limit', :attachment_processing do
+    it 'is set differently whether vips is enabled or not' do
+      expect(described_class::LIMIT).to eq (Rails.configuration.x.use_vips ? 8.megabytes : 2.megabytes)
+    end
+  end
```

Comment:
I think this is akin to testing a method using its own implementation.

Unless you can test how the constant changes when you change the `use_vips` configuration (and I think you cannot, since it is a constant), I believe this test to have little value.

Reply:
Sorry I did not make that more explicit, but the reason for this test is just to ensure the constant is properly set regardless of load order.

Indeed, the constant is defined at module load-time and I'm not entirely confident nothing will ever cause `PreviewCard` to be loaded before the initializer that sets `Rails.configuration.x.use_vips` changes.

I guess I could also have made limit a `Proc` checking `Rails.configuration.x.use_vips` as runtime, but it would have required to change the definition of `remotable_attachment` and possibly Paperclip (haven't checked if Paperclip takes a proc as `limit`)

Reply:
Paperclip should support a `Proc` as `less_than`, I can switch to that  route if you prefer.

Reply:
I am not a fan of needlessly complicating application code just to improve "testability".

I was actually suggesting just removing the test altogether :sweat_smile: 

But I already approved the changes, so I leave it up to you.

Reply:
Ah no, this is not to improve the test but to improve confidence in the correct value being used. As this currently depends on which order modules are loaded (and the test is here to ensure that) and the proposed change would make it so that it doesn't depend on load order anymore.